themes/src/main/resources/theme/base/login/resources/js/authChecker.js

```diff
@@ -0,0 +1,76 @@
+
+function kcAuthChecker() {
+    var authChecker = this;
+
+    // How often to check if KEYCLOAK_SESSION cookie was added in the other browser tab?
+    var checkIntervalMillisecs = 2000;
+
+    // 1 - unknown state or KEYCLOAK_SESSION present since beginning, 2 - cookie KEYCLOAK_SESSION not present, 3 - cookie KEYCLOAK_SESSION present after it
+    // was not present before (possible to move here from state 2, but not from 1. Reason are "re-authentication" scenarios where we want login screens displayed even if SSO cookie exists)
+    var sessionCookieState = 1;
```

Comment:
Rather than keeping this in a rather magical number, we can instead define an object with a property for each of the states. JavaScript doesn't have enums, so this is the closes approximation we can make:

```js
const SessionState = {
  /** Unknown state, or the session was present since the beginning. */
  Unknown: "UNKNOWN",
  /** Session is not present. */
  NotPresent: "NOT_PRESENT",
  /** Session was not present before, but it's present now. */
  Present: "PRESENT",
};
```

We can then do things like assign that status to a variable:

```js
const sessionState = SessionState.Unknown;
```

And make comparisons at later points:

```js
if (sessionState === SessionState.Present) {
  // Do something...
}
```